server/src/main/java/org/apache/druid/server/metrics/BrokerSegmentCountStatsMonitor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.metrics;
+
+import com.google.inject.Inject;
+import org.apache.druid.java.util.emitter.service.ServiceEmitter;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.apache.druid.java.util.metrics.AbstractMonitor;
+import org.apache.druid.server.coordinator.stats.RowKey;
+
+import java.util.Map;
+
+/**
+ * Monitor that tracks the number of segments of a datasource currently queryable by this Broker.
+ */
+public class BrokerSegmentCountStatsMonitor extends AbstractMonitor
+{
+  private final BrokerSegmentCountStatsProvider statsProvider;
+
+  @Inject
+  public BrokerSegmentCountStatsMonitor(BrokerSegmentCountStatsProvider statsProvider)
+  {
+    this.statsProvider = statsProvider;
+  }
+
+  @Override
+  public boolean doMonitor(ServiceEmitter emitter)
+  {
+    emit(emitter, "segment/available/count", statsProvider.getAvailableSegmentCount());
+    return true;
+  }
+
+  private void emit(ServiceEmitter emitter, String key, Map<RowKey, Long> counts)
```

Comment:
Should we move this method to the `AbstractMonitor` class? Other monitor classes also seem to have a similar logic.